### PR TITLE
Ensure incremental bulk setting is set atomically

### PR DESCRIPTION
**Pull request by Tim-Brooks**
Currently the rest.incremental_bulk is read in two different places.
This means that it will be employed in two steps introducing
unpredictable behavior. This commit ensures that it is only read in a
single place.

### CODE REVIEW

**Review by mhl-b — APPROVED** [review]
lgtm

File: server/src/main/java/org/elasticsearch/action/bulk/IncrementalBulkService.java [hunk 1]
```diff
@@ -14,74 +14,72 @@
...
         return new Handler(client, threadContext, indexingPressure, waitForActiveShards, timeout, refresh);
     }
 
+    private void ensureEnabled() {
+        if (enabled.get() == false) {
+            throw new AssertionError("Unexpected incremental bulk request");
+        }
+    }
+
+    // This method only exists to tests that the feature flag works. Remove once we no longer need the flag.
```
Comment (henningandersen) [inline server/src/main/java/org/elasticsearch/action/bulk/IncrementalBulkService.java:69]: Can we put a similar comment on the `enabled` field (or rename it to `enabledForTests`)?

File: server/src/main/java/org/elasticsearch/action/bulk/IncrementalBulkService.java [hunk 2]
```diff
@@ -14,74 +14,72 @@
...
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.function.Supplier;
 
+import static org.elasticsearch.common.settings.Setting.boolSetting;
+
 public class IncrementalBulkService {
 
+    public static final Setting<Boolean> INCREMENTAL_BULK = boolSetting(
+        "rest.incremental_bulk",
+        true,
```
Comment (henningandersen) [inline server/src/main/java/org/elasticsearch/action/bulk/IncrementalBulkService.java:38]: I'd probably want to flip this before merge to main? Unless we are confident in having the feature enabled without a rollout plan. Can be a follow-up.

**Review by henningandersen — APPROVED** [review]
LGTM.

File: server/src/main/java/org/elasticsearch/action/bulk/IncrementalBulkService.java (hunk 2 above)
Comment (Tim-Brooks) [reply server/src/main/java/org/elasticsearch/action/bulk/IncrementalBulkService.java:38]: I will add a ticket and make it a blocker for merge into main.

**Pull request merged**